tests/ignite/handlers/test_checkpoint.py

```diff
@@ -274,6 +269,45 @@ def _test(to_save, obj, name):
     _test(to_save, {"model": model.state_dict(), "optimizer": optimizer.state_dict()}, "checkpoint")
 
 
+def test_checkpoint_with_score_name_only():
```

Comment:
Can you put this test into test_checkpoint_with_score_name_and_function as a special case of function as None

Reply:
To get the score_function using score_name that requires the metric to be attached, that will not be a good change in this test, so I suggest creating two separate tests for these 2 cases